server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -2401,7 +2401,7 @@ private void deleteSnapshotsFromRepository(
     ) {
         if (repositoryOperations.startDeletion(deleteEntry.uuid())) {
             assert currentlyFinalizing.contains(deleteEntry.repository());
-            final List<SnapshotId> snapshotIds = deleteEntry.getSnapshots();
+            final Set<SnapshotId> snapshotIds = new HashSet<>(deleteEntry.getSnapshots());
```

Comment:
This is really subtle now, I think if we want to optimize this we should either make the signature of `withRemovedSnapshots` take a set or copy to a set in that method. Making the signature a set is probably cleanest

Reply:
I agree. I would also prefer to just copy the `snapsthotIds` inside `withRemovedSnapshots` for now, I do not think it really matters that we copy it there?

Reply:
I don't think performance here matters that much, that method is not running all that often either. Lets keep it simple IMO